modules/adding-ibmz-lpar-agent.adoc

```diff
@@ -0,0 +1,105 @@
+<4> Specify this parameter when you use an Open Systems Adapter (OSA) or HiperSockets.
+
+. Generate the `.ins` and `initrd.img.addrsize` files.
++
+The `.ins` file is a special file that includes installation data and is present on the FTP server. It can be accessed from the HMC system.
++
+This file contains details such as mapping of the location of installation data on the disk or FTP server, the memory locations where the data is to be copied.
++
+*Note*: The `.ins` file and `initrd.img.addrsize` are not automatically generated as part of boot-artifacts from the installer. You must manually generate these files.
+
+.. Run the following commands to get the size of the `kernel` and `initrd`: 
++
+[source,terminal]
+----
+KERNEL_IMG_PATH='./kernel.img'
+INITRD_IMG_PATH='./initrd.img'
+CMDLINE_PATH='./generic.prm'
+kernel_size=$(stat -c%s $KERNEL_IMG_PATH )
+initrd_size=$(stat -c%s $INITRD_IMG_PATH)
+----
+.. Round the `kernel` size up to the next MB boundary. This value  is the starting address of `initrd.img`.
++
+[source,terminal]
+----
+offset=$(( (kernel_size + 1048575) / 1048576 * 1048576 ))
+----
+.. Create the kernel binary patch file that contains the `initrd` address and size by running the following commands:
++
+[source,terminal]
+----
+INITRD_IMG_NAME=$(echo $INITRD_IMG_PATH | rev | cut -d '/' -f 1 | rev)
+KERNEL_OFFSET=0x00000000
+KERNEL_CMDLINE_OFFSET=0x00010480
+INITRD_ADDR_SIZE_OFFSET=0x00010408
+OFFSET_HEX=$(printf '0x%08x\n' $offset)
+----
+
+.. Convert the address and size to binary format by running the following commands:
++
+[source,terminal]
+----
+printf "$(printf '%016x\n' $initrd_size)" | xxd -r -p > temp_size.bin
+----
+
+.. Concatenate the address and size binaries by running the following command:
++
+[source,terminal]
+----
+cat temp_address.bin temp_size.bin > "$INITRD_IMG_NAME.addrsize"
+----
+
+.. Clean up temporary files by running the following command:
++
+[source,terminal]
+----
+rm -rf temp_address.bin temp_size.bin
+----
+
+.. Create the `.ins` file. The file is based on the paths of the `kernel.img`, `initrd.img`, `initrd.img.addrsize`, and `cmdline` files and the memory locations where the data is to be copied.
++
+[source,terminal]
+----
+$KERNEL_IMG_PATH $KERNEL_OFFSET
+$INITRD_IMG_PATH $OFFSET_HEX
+$INITRD_IMG_NAME.addrsize $INITRD_ADDR_SIZE_OFFSET
+$CMDLINE_PATH $KERNEL_CMDLINE_OFFSET
+----
+
+. Transfer the `initrd`, `kernel`, `generic.ins`, and `initrd.img.addrsize` parameter files to the file server. For more information about how to transfer the files with FTP and boot, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/performing_a_standard_rhel_8_installation/installing-in-an-lpar_installing-rhel[Installing in an LPAR].    
```

Comment:
This link brings me to the RHEL 8 homepage. Was this intended?

![image](https://github.com/user-attachments/assets/94e7de78-99bf-4837-86aa-d549df97af35)

Reply:
Seems like the RHEL docs have been reorganized we also have that link in the UPI docs it worked a few months ago. I'll check where we can link to.

Reply:
@krishnanshruthi Seems like the entire IBM Z documentation has vanished from the RHEL docs. Please take this link instead: https://www.ibm.com/docs/en/linux-on-systems?topic=bl-booting-linux-in-lpar-mode

Reply:
@alishaIBM I don't see IBM Power in the RHEL docs anymore either. Just in case you also have links to RHEL

Reply:
The link still points to the RHEL CP homepage but this is OK?